src/scene/lightmapper.js

```diff
@@ -446,10 +489,10 @@ Object.assign(Lightmapper.prototype, {
         var casters = [];
         var meshes;
         for (node = 0; node < allNodes.length; node++) {
-            origCastShadows[node] = allNodes[node].model.castShadows;
-            allNodes[node].model.castShadows = allNodes[node].model.castShadowsLightmap;
-            if (allNodes[node].model.castShadowsLightmap) {
-                meshes = allNodes[node].model.meshInstances;
+            origCastShadows[node] = renderOrModel(allNodes[node]).castShadows;
+            renderOrModel(allNodes[node]).castShadows = renderOrModel(allNodes[node]).castShadowsLightmap;
+            if (renderOrModel(allNodes[node]).castShadowsLightmap) {
+                meshes = renderOrModelMeshInstances(allNodes[node]);
```

Comment:
Style: We can cache `renderOrModel()` over calling it multiple times here.

Reply:
done